x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/DelegatePkiAuthenticationResponse.java

```diff
@@ -69,10 +59,15 @@ public TimeValue getExpiresIn() {
         return expiresIn;
     }
 
+    public Authentication getAuthentication() {
+        return authentication;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(accessToken);
         out.writeTimeValue(expiresIn);
+        authentication.writeTo(out);
```

Comment:
This needs to be guarded by version check as well, e.g.
```java
if (out.getVersion().onOrAfter(..) {
}
```

Reply:
Fixed